### PR TITLE
Allow calling managed methods from objects bound with RegisterAsyncJsObject that return arrays of structs

### PR DESCRIPTION
**Pull request by joaompneves**
**Problem:**
It was not possible to call .Net methods from JS that return arrays of structs.

Fix:
IsComplexType from JavascriptObjectRepository returns false for arrays.

Notes:
This PR requires [1979](https://github.com/cefsharp/CefSharp/pull/1979)

### CODE REVIEW

**Comment by amaitland** [comment]
> Make it happen :)

A technical description please

This `PR` also contains #1979, please either only include the changes relevant to this fix or add all the relevant technical details to this `PR` please.

File: CefSharp.Example/AsyncBoundObject.cs [hunk 1]
```diff
@@ -33,5 +37,12 @@ public void DoSomething()
         {
             Thread.Sleep(1000);
         }
+
+        public JsObject[] ObjectArray(string name) {
```
Comment (amaitland) [inline CefSharp.Example/AsyncBoundObject.cs]: Formatting please, braces on `new lines, be consistent with other

File: CefSharp.Example/AsyncBoundObject.cs [hunk 2]
```diff
@@ -8,6 +8,10 @@
 
 namespace CefSharp.Example
 {
+    public struct JsObject {
```
Comment (amaitland) [inline CefSharp.Example/AsyncBoundObject.cs]: Brace on new line please

File: CefSharp.Example/Resources/BindingTest.html [hunk 3]
```diff
@@ -82,11 +82,21 @@
                         writeAsyncResult(call, end);
                     });
                 }
+
+                function asyncObjectArray() {
```
Comment (amaitland) [inline CefSharp.Example/Resources/BindingTest.html]: Braces on new lines

File: CefSharp.Example/Resources/BindingTest.html [hunk 4]
```diff
@@ -82,11 +82,21 @@
...
+                        var end = "Result: [ " + res.map(function (item) { return item.Value }) + " ] (" + Date() + ")";
+                        writeAsyncResult(call, end);
+                    });
+                }
             
                 asyncError();
                 asyncDivOk();
                 asyncDivFail();
                 asyncDoSomething();
+                asyncHello();
```
Comment (amaitland) [inline CefSharp.Example/Resources/BindingTest.html]: Is there an `asyncHello` method?

Comment (joaompneves) [inline CefSharp.Example/Resources/BindingTest.html]: Yes

**Pull request merged**